app/src/main/java/fi/aalto/legroup/achso/entities/UserPool.java

```diff
@@ -0,0 +1,55 @@
+package fi.aalto.legroup.achso.entities;
+
+import java.util.ArrayList;
+
+/**
+ * Use this class to store users memory efficiently.
+ * Maps User objects to indices that can later retrieve the actual User back.
+ */
+public class UserPool {
```

Comment:
Well it _could_ be but with the new index references it's probably too dangerous for _public_ consumption, making package-private.